crates/faucet/src/lib.rs

```diff
@@ -180,13 +180,68 @@ impl Faucet {
         Ok(Self {
             id,
             client,
+            store: sqlite_store,
+            grpc_client,
             tx_prover,
             issuance,
             max_supply,
             script,
         })
     }
 
+    /// Syncs the state of the client.
+    #[instrument(target = COMPONENT, name = "faucet.sync_state", skip_all, err)]
+    async fn sync_state(
+        account_id: AccountId,
+        store: Arc<dyn Store>,
+        grpc_client: Arc<dyn NodeRpcClient>,
+    ) -> anyhow::Result<SyncSummary> {
+        let note_screener = NoteScreener::new(store.clone());
+        let state_sync = StateSync::new(grpc_client, Arc::new(note_screener), None);
+
+        // Get current state of the client
+        let accounts = store
+            .get_account_header(account_id)
+            .await?
+            .map(|(header, _)| vec![header])
+            .unwrap_or_default();
+        let note_tags: BTreeSet<NoteTag> = store.get_unique_note_tags().await?;
+
+        let unspent_input_notes = vec![];
+        let unspent_output_notes = vec![];
```

Comment:
Here, I think we do want to update notes that were not committed:


```suggestion
        let expected_output_notes = store.get_output_notes(OutputNoteFilter::Expected).await?;
```